app/controllers/finance/lead_providers_controller.rb

```diff
@@ -1,7 +1,34 @@
 # frozen_string_literal: true
 
 class Finance::LeadProvidersController < Finance::BaseController
+  skip_after_action :verify_policy_scoped
+
   def index
-    @lead_providers = LeadProvider.all
+    @ecf_lead_providers = LeadProvider.all
+  end
+
+  def show
+    @ecf_lead_provider = lead_provider_scope.find(params[:id])
+    cpd_lead_provider = @ecf_lead_provider.cpd_lead_provider
+
+    @breakdown = CalculationOrchestrator.call(
+      cpd_lead_provider: cpd_lead_provider,
+      contract: @ecf_lead_provider.call_off_contract,
+      aggregator: ::ParticipantEventAggregator,
+      uplift_aggregator: ::ParticipantUpliftAggregator,
+      calculator: ::PaymentCalculator::Ecf::PaymentCalculation,
+      event_type: :started,
+    )
+
+    @total_ect = ParticipantEventAggregator.call({ cpd_lead_provider: cpd_lead_provider, started: :count_active_ects_for_lead_provider })
+    @total_mentors = ParticipantEventAggregator.call({ cpd_lead_provider: cpd_lead_provider, started: :count_active_mentors_for_lead_provider })
+    @total_participants = ParticipantEventAggregator.call({ cpd_lead_provider: cpd_lead_provider, started: :count_active_for_lead_provider })
```

Comment:
I'm looking at the code for `ParticipantEventAggregator` and I have a hard time understanding what's going on in there.

```
ParticipantEventAggregator.call({ cpd_lead_provider: cpd_lead_provider, started: :count_active_ects_for_lead_provider })

...

class ParticipantEventAggregator
  include HasDIParameters

  def call(event_type: :started)
    recorder.send(params[event_type], cpd_lead_provider)
  end

private

  def default_params
    {
      recorder: ParticipantDeclaration,
      started: :count_active_for_lead_provider,
    }
  end
end
```
If I see this correctly... `ParticipantEventAggregator.call` makes a new instance of `ParticipantEventAggregator`, and `call`s it with the same parameters. Which the instance ignores, because the only thing its call cares about is `event_type`. It gets all other things from fields that were created in its `new` call. 
In the meantime, the params with which we called the class need to include the hash for `event_type` value, and the value of that hash needs to be a key matching a scope on a model somewhere. 

Other complexities aside, I think it is not a great idea to have our `new` and `call` methods take the same set of arguments - because at best those sets are independent, which means both methods receive and ignore a lot of needless arguments, and at worst they aren't independent, and you can't `call` with a different value of a param than you `new` - unless you break the general pattern...

Not sure if this is the right PR to fix it in, probably not :(

Reply:
https://github.com/DFE-Digital/early-careers-framework/blob/9a5480476680773d0964d8f7ddd2e540c8a20937/app/services/participant_event_aggregator.rb :eyes:
Perhaps @blackrat would be best placed to respond to this one